Homework/Nam_Nguyen/Lesson1/LeetCode_304_RangeOfQuery2DImmutable.java

```diff
@@ -0,0 +1,36 @@
+package Homework.Nam_Nguyen.Lesson1;
+
+public class LeetCode_304_RangeOfQuery2DImmutable {
+
+    private int[][] sumArray;
+
+    public LeetCode_304_RangeOfQuery2DImmutable(int[][] matrix) {
+        int rows = matrix.length;
+        int columns = matrix[0].length;
+        sumArray = new int[rows][columns];
+        sumArray[0][0] = matrix[0][0];
+        for (int i = 0; i < rows; i++) {
+            sumArray[i][0] = matrix[i][0];
+            for (int j = 1; j < columns; j++) {
+                sumArray[i][j] = sumArray[i][j - 1] + matrix[i][j];
+            }
+        }
+    }
+
+    public int sumRegion(int row1, int col1, int row2, int col2) {
```

Comment:
@namnp9981 can try improve this to O(1)?

Reply:
Ok, I'm trying to find out a better way to improve this.